pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +501,168 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
 		return generatedCodeSoFar;
 	}
 
+
+	/**
+	 * Generate the compareTo method content for the provided fields
+	 * @param fields The fields of this object.
+	 * @param generatedCodeSoFar the generated code so far (non-empty in case of nested objects)
+	 * @return The generated code for compareTo method body
+	 */
+	public static String getFieldsCompareToStatements(final List<Field> fields, String generatedCodeSoFar) {
+		for(Field f : fields) {
```

Comment:
This codebase would probably benefit from running the same spotless rules as the hedera-services codebase for consistency.
Perhaps Nathan can help get that setup for this repo?

Reply:
@jsync-swirlds I tried applying spotless rules here, and that resulted in a disaster with indentations in the generated code. So, I don't think it's going to be that straightforward.

Reply:
@anthony-swirldslabs 

> we do expect the fields list to be in the exact order in which the fields are declared in the model definition, right?

It's not necessarily an expectation. It just happened to be this way, it's a reasonable default. As long as this standard applies consistently (and it always should the case), there should be no issues
However, you raised a good point. What if we have predefined set of fields we want to use in `compareTo`? Right now the order is not preserved, and it's not good. It should be.

Reply:
Regarding spotless, it's not setup on this repo currently.  I would not expect it to work until that is done (which probably requires assistance from Nathan or another member of that team).

I am just suggesting that (aside from this PR) it might be worthwhile to get that effort on the priority list so it can be done at some point.

Reply:
> @jsync-swirlds I tried applying spotless rules here, and that resulted in a disaster with indentations in the generated code. So, I don't think it's going to be that straightforward.

Ya, we should do it as part of a separate PR. But it would be nice to get spotless going.

Reply:
Okay, I'll try make it work in a separate PR

Reply:
Perhaps just open an issue for now?  It will probably take some effort, and isn't strictly related to the issue driving this PR.